Thunder/utils/rate_limiter.py

```
@@ -0,0 +1,425 @@
+                        self.average_processing_time = sum(self.processing_times) / len(self.processing_times)
+
+                    file_identifier = request_data.get('file_identifier')
+                    if file_identifier:
+                        file_times = self.file_processing_times.setdefault(file_identifier, deque(maxlen=100))
+                        file_times.append(processing_time)
+
+                except FloodWait as e:
+                    logger.warning(f"FloodWait for user {user_id}, waiting {e.value}s before re-queuing.")
+                    await asyncio.sleep(e.value)
+                    await self._requeue_request(request_data, queue_type)
+                except Exception as e:
```

Ruff (0.13.1)

Do not catch blind exception: `Exception`

(BLE001)